### PR TITLE
Fix phantom read

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

File: store/localstore/kv.go [hunk 1]
```diff
@@ -140,17 +150,20 @@ func (s *dbStore) CurrentVersion() (kv.Version, error) {
 
 // Begin transaction
 func (s *dbStore) Begin() (kv.Transaction, error) {
+	lockVersionProvider()
+	beginVer, err := globalVersionProvider.CurrentVersion()
+	if err != nil {
```
Comment (siddontang) [inline store/localstore/kv.go]: you can use unlockVersionProvider here and then check err.

File: store/localstore/isolation_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+				err = kv.RunInNewTxn(store, true, func(txn kv.Transaction) error {
+					var err1 error
+					id, err1 = txn.Inc([]byte("key"), 1)
+					return err1
+				})
+				c.Assert(err, IsNil)
+
+				m.Lock()
+				_, ok := ids[id]
+				c.Assert(ok, IsFalse)
```
Comment (siddontang) [inline store/localstore/isolation_test.go]: can't assert here, in my test, if assert here, if failed, the whole test will be hung up.

can use below: 
```
m.Lock()
_, ok := ids[id]
ids[id] = struct{}{}
m.Unlock()
c.Assert(ok, IsFalse)
```

File: store/localstore/isolation_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+)
+
+var _ = Suite(&testIsolationSuite{})
+
+type testIsolationSuite struct {
+	s kv.Storage
+}
+
+func (t *testIsolationSuite) TestInc(c *C) {
+	store, err := tidb.NewStore("memory://test_ddl/test_ddl")
```
Comment (siddontang) [inline store/localstore/isolation_test.go]: test_ddl -> test may be better.
The test is not for test_ddl.

File: store/localstore/isolation_test.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	defer store.Close()
+
+	s, err := tidb.CreateSession(store)
```
Comment (siddontang) [inline store/localstore/isolation_test.go]: no need to CreateSession, can remove it.

**Comment by shenli** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**